mender-convert-modify

```diff
@@ -233,78 +233,73 @@ EOF
-
-            log_info "Installing Mender Flash version ${MENDER_FLASH_VERSION}"
-            deb_get_and_install_package mender-flash mender-flash "${MENDER_FLASH_VERSION}"
-
-            if [ "$MENDER_SETUP_INSTALL" = "auto" ]; then
-                MENDER_SETUP_INSTALL=y
-            fi
-
-            if [ "$MENDER_SNAPSHOT_INSTALL" = "auto" ]; then
-                MENDER_SNAPSHOT_INSTALL=y
-            fi
-            ;;
-    esac
+    mender_client_deb_version="$MENDER_CLIENT_VERSION"
+
+    log_info "Installing Mender Client version ${mender_client_deb_version}"
+    pkgs_to_install["mender-client4"]="$mender_client_deb_version"
 
-    if [ "$MENDER_SETUP_INSTALL" = "y" ]; then
+    log_info "Installing Mender Flash version ${MENDER_FLASH_VERSION}"
+    pkgs_to_install["mender-flash"]="${MENDER_FLASH_VERSION}"
+
+    if [ "$MENDER_SETUP_INSTALL" = "auto" ] || [ "$MENDER_SETUP_INSTALL" = "y" ]; then
         log_info "Installing Mender Setup version ${MENDER_SETUP_VERSION}"
-        deb_get_and_install_package mender-setup mender-setup "${MENDER_SETUP_VERSION}"
+        pkgs_to_install["mender-setup"]="${MENDER_SETUP_VERSION}"
     fi
 
-    if [ "$MENDER_SNAPSHOT_INSTALL" = "y" ]; then
+    if [ "$MENDER_SNAPSHOT_INSTALL" = "auto" ] || [ "$MENDER_SNAPSHOT_INSTALL" = "y" ]; then
         log_info "Installing Mender Snapshot version ${MENDER_SNAPSHOT_VERSION}"
-        deb_get_and_install_package mender-snapshot mender-snapshot "${MENDER_SNAPSHOT_VERSION}"
+        pkgs_to_install["mender-snapshot"]="${MENDER_SNAPSHOT_VERSION}"
     fi
-
-    # Save installed client version for tests in Yocto variable format
-    testscfg_add "PREFERRED_VERSION:mender-client" "$(echo ${mender_client_deb_name} | sed -r 's/.*_([0-9]+\.[0-9]+\.[0-9]+).*/\1/')"
 fi
 
 if [ "${MENDER_ADDON_CONNECT_INSTALL}" = "y" ]; then
     log_info "Installing Mender Connect addon"
-    deb_get_and_install_package mender-connect mender-connect "${MENDER_ADDON_CONNECT_VERSION}"
+    pkgs_to_install["mender-connect"]="${MENDER_ADDON_CONNECT_VERSION}"
 fi
 
 if [ "${MENDER_ADDON_CONFIGURE_INSTALL}" = "y" ]; then
     log_info "Installing Mender Configure addon"
-    deb_get_and_install_package mender-configure mender-configure "${MENDER_ADDON_CONFIGURE_VERSION}" "true"
+    pkgs_to_install["mender-configure"]="${MENDER_ADDON_CONFIGURE_VERSION}"
+fi
+
+if [[ ${#pkgs_to_install[@]} -gt 0 ]]; then
+    deb_ensure_repo_enabled
+    if echo ${pkgs_to_install[@]} | grep -qF "master"; then
+        deb_ensure_repo_enabled "experimental"
+    fi
+
+    # Collect exact package versions to install
+    pkgs_with_versions=""
+    for pkg in ${!pkgs_to_install[@]}; do
+        pkgs_with_versions="${pkgs_with_versions} $(deb_get_repo_package "$pkg" "${pkgs_to_install[$pkg]}")"
+    done
+
+    # Save installed client version for tests in Yocto variable format
+    mender_client_version="$(echo ${pkgs_with_versions} | sed -r 's/.*mender_client4=([0-9]+\.[0-9]+\.[0-9]+).*/\1/')"
```

Comment:
could you double check: this will mean that if the pkgs_with_verrsions does not have something like "mender_client4=1.1.1" the whole `${pkgs_with_versions}` will endup in the `mender_client_version`